### DISCUSSION TITLE
Where is the List<T> implementation?

### DISCUSSION THREAD

**Discussion by EdwinCloud101** [dotnet/runtime#41564] [General] [answered]
I see the class from System.Collections.cs but I don't see any code at all. Am I missing something silly here?

thanks

![image](https://user-images.githubusercontent.com/2415262/91650167-3c65b780-ea4a-11ea-965d-732b41642176.png)

**Comment by reflectronic** [comment, marked as answer]
You are looking at the *reference assembly* for `System.Collections`. The reference assembly lives under the `ref` directory and does not actually provide any code for the classes. The source for the *implementation assembly*, which will actually contain the code, lives under the `src` directory. You can read more about what reference assemblies are used for in the [documentation](https://docs.microsoft.com/en-us/dotnet/standard/assembly/reference-assemblies).

Even though `List<T>` is defined in the reference assembly for `System.Collections`, it does not actually live in the implementation assembly for `System.Collections`. It actually lives in `System.Private.CoreLib`, which is the "root assembly" where types like `System.Object` are defined. You can find the file for List<T> at `\src\libraries\System.Private.CoreLib\src\System\Collections\Generic\List.cs`.

**Comment by gfoidl** [comment]
Side note: .NET Source Browser is a helpful web-app for this --> https://source.dot.net/#q=List%3CT%3E
